src/containers/ArticlePage/TopicArticlePage/components/TopicArticleConnections.tsx

```diff
@@ -0,0 +1,192 @@
+    fetchFavoriteSubjects();
+  }, []);
+
+  const fetchFavoriteSubjects = async () => {
+    const result = await fetchUserData();
+    const favoriteSubjects = result.favoriteSubjects || [];
+    setFavoriteSubjectIds(favoriteSubjects);
+  };
+
+  const getFavoriteSubjects = (subjects: SubjectType[], favoriteSubjectIds: string[]) => {
+    return subjects.filter(e => favoriteSubjectIds.includes(e.id));
+  };
+
+  const handleOpenToggle = async ({
+    path,
+    isSubject,
+    id,
+  }: {
+    path: string;
+    isSubject: boolean;
+    id: string;
+  }) => {
+    const { allowMultipleSubjectsOpen, getSubjectTopics } = props;
+    let paths = [...openedPaths];
+    const index = paths.indexOf(path);
+    if (index === -1) {
+      if (isSubject) {
+        getSubjectTopics(id, props.locale);
+        if (!allowMultipleSubjectsOpen) {
+          paths = [];
+        }
+      }
+      paths.push(path);
+    } else {
+      paths.splice(index, 1);
+    }
+    setOpenedPaths(paths);
+  };
+
+  const addTopic = async (path: string, closeModal: () => void) => {
+    props.stageTaxonomyChanges({ path });
+    closeModal();
+  };
+
+  const toggleShowFavorites = () => {
+    setShowFavorites(!showFavorites);
+  };
+  const { t, structure, activeTopics, ...rest } = props;
+
+  return (
+    <Fragment>
+      <FieldHeader
+        title={t('taxonomy.topics.topicPlacement')}
+        subTitle={t('taxonomy.topics.subTitleTopic')}>
+        <HowToHelper pageId="TaxonomyTopicConnections" tooltip={t('taxonomy.topics.helpLabel')} />
+      </FieldHeader>
+      <ActiveTopicConnections activeTopics={activeTopics} type="topic-article" {...rest} />
+      <Modal
+        backgroundColor="white"
+        animation="subtle"
+        size="large"
+        narrow
+        minHeight="85vh"
+        activateButton={<Button>{t(`taxonomy.topics.${'chooseTaxonomyPlacement'}`)}</Button>}>
+        {(closeModal: () => void) => (
+          <Fragment>
+            <ModalHeader>
+              <ModalCloseButton
+                title={t('taxonomy.topics.filestructureClose')}
+                onClick={closeModal}
+              />
+            </ModalHeader>
+            <ModalBody>
+              <ModalTitleRow>
+                <StyledTitleModal>{t('taxonomy.topics.filestructureHeading')}:</StyledTitleModal>
+                <Switch
+                  onChange={toggleShowFavorites}
+                  checked={showFavorites}
+                  label={t('taxonomy.favorites')}
+                  id={'favorites'}
+                />
+              </ModalTitleRow>
+              <hr />
+              <Structure
+                openedPaths={openedPaths}
+                structure={
+                  showFavorites ? getFavoriteSubjects(structure, favoriteSubjectIds) : structure
+                }
+                toggleOpen={handleOpenToggle}
+                renderListItems={(props: any) => {
+                  // item should possibly be typed in [Structure].
```

Comment:
hehe ja, her må du finne ut av typen, re det samme props som for komponenten så kan du kanskje bruke Props